src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -0,0 +1,170 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""A data window that moves with the latest datapoints of a data stream."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+from collections.abc import Sequence
+from datetime import datetime, timedelta
+from typing import Union
+
+import numpy as np
+from frequenz.channels import Receiver
+from numpy.typing import ArrayLike
+
+from frequenz.sdk.timeseries import Sample
+from frequenz.sdk.timeseries._ringbuffer import OrderedRingBuffer
+
+log = logging.getLogger(__name__)
+
+
+class MovingWindow(Sequence):
+    """
+    A data window that moves with the latest datapoints of a data stream.
+
+    After initialization the `MovingWindow` can be accessed by an integer
+    index or a timestamp. A sub window can be accessed by using a slice of integers
+    integers or timestamps.
+
+    Note that a numpy ndarray is returned and thus users can use
+    numpys operations directly on a window.
+
+    **Example1** (calculating the mean of a time interval):
+
+    ```
+    window = MovingWindow(size = 100, resampled_data_recv = resampled_data_recv)
```

Comment:
Silly and this shows that maybe we also need to run `black` on the documentation code samples too :laughing: , but it should be:

```suggestion
    window = MovingWindow(size=100, resampled_data_recv=resampled_data_recv)
```